lib/sax.js

```diff
@@ -651,16 +649,22 @@ ElementAttributes.prototype = {
 	//	getType:function(i){},
 };
 
-function split(source, start) {
-	var match;
-	var buf = [];
-	var reg = /'[^']+'|"[^"]+"|[^\s<>\/=]+=?|(\/?\s*>|<)/g;
-	reg.lastIndex = start;
-	reg.exec(source); //skip <
-	while ((match = reg.exec(source))) {
-		buf.push(match);
-		if (match[1]) return buf;
+function _experimentalDoctypeRegex(source, start) {
+	// since 0.5.0 we purposefully do not remove the quotes from publicId and systemId
+	// so the values can be serialized as is
+	var regex =
+		/<!DOCTYPE\s+(([:A-Z_a-z\u00C0-\u00D6\u00D8-\u00F6\u00F8-\u02FF\u0370-\u037D\u037F-\u1FFF\u200C-\u200D\u2070-\u218F\u2C00-\u2FEF\u3001-\uD7FF\uF900-\uFDCF\uFDF0-\uFFFD]|[\uD800-\uDB7F\uDC00-\uDFFF])([:A-Z_a-z\u00C0-\u00D6\u00D8-\u00F6\u00F8-\u02FF\u0370-\u037D\u037F-\u1FFF\u200C-\u200D\u2070-\u218F\u2C00-\u2FEF\u3001-\uD7FF\uF900-\uFDCF\uFDF0-\uFFFD\-\.0-9\u00B7\u0300-\u036F\u203F-\u2040]|[\uD800-\uDB7F\uDC00-\uDFFF])*)(\s+(SYSTEM)\s+("[^"]*"|'[^']*')|\s+(PUBLIC)\s+("[\r\n\x20\w\-'()+,.\/:=?;!*#@$_%]*"|'[\r\n\x20\w\-()+,.\/:=?;!*#@$_%]*')\s+("[^"]*"|'[^']*'))?\s*(\[((\s*(%(([:A-Z_a-z\u00C0-\u00D6\u00D8-\u00F6\u00F8-\u02FF\u0370-\u037D\u037F-\u1FFF\u200C-\u200D\u2070-\u218F\u2C00-\u2FEF\u3001-\uD7FF\uF900-\uFDCF\uFDF0-\uFFFD]|[\uD800-\uDB7F\uDC00-\uDFFF])([:A-Z_a-z\u00C0-\u00D6\u00D8-\u00F6\u00F8-\u02FF\u0370-\u037D\u037F-\u1FFF\u200C-\u200D\u2070-\u218F\u2C00-\u2FEF\u3001-\uD7FF\uF900-\uFDCF\uFDF0-\uFFFD\-\.0-9\u00B7\u0300-\u036F\u203F-\u2040]|[\uD800-\uDB7F\uDC00-\uDFFF])*);|<!ELEMENT ([^>]*)>|<!ATTLIST ([^>]*)>|<!ENTITY ([^>]*)>|<!NOTATION ([^>]*)>|<\?([^>]*)\?>|<!--([^>]*)-->)\s*)*)])?\s*>/;
+	var match = regex.exec(source.slice(start));
+	if (!match) {
+		throw new Error('Parse Error: malformed document type');
```

Comment:
inside the sax parser all error should be reported using the `errorHandler`, in this case I would say it is at least an `error` if not a `fatalError`. I need to look up which one it is, since there is a definition for it. I assume this would be an issue of "not well-formed".
In case of the `fatalError` it would would prevent any further processing.